lib/pbench/test/unit/server/query_apis/test_datasets_publish.py

```diff
@@ -187,8 +187,36 @@ def test_no_dataset(
         assert response.status_code == HTTPStatus.NOT_FOUND
         assert response.json["message"] == "Dataset 'badwolf' not found"
 
+    def test_no_index(
+        self, client, monkeypatch, attach_dataset, pbench_token, server_config
+    ):
+        """
+        Check the publish API if the dataset has no INDEX_MAP. It should
+        fail with a CONFLICT error.
+        """
+        self.fake_elastic(monkeypatch, {}, True)
+
+        ds = Dataset.query(name="drb")
+        response = client.post(
+            f"{server_config.rest_uri}/datasets/publish/{ds.resource_id}",
+            headers={"authorization": f"Bearer {pbench_token}"},
+            json=self.PAYLOAD,
+        )
+
+        # Verify the report and status
+        assert response.status_code == HTTPStatus.CONFLICT
+        assert response.json == {
+            "message": "Dataset update requires Indexed state (INDEXED)"
+        }
```

Comment:
Is this message telling us that it failed because, instead of being in the "Indexed" state, it was in the "INDEXED" state?  😀

Reply:
Yeah, I saw that; of course it's an artifact of the fake dataset in the fixture, and it looks weird, but for the setup it's correct.